newIDE/app/src/UI/Accordion.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+import * as React from 'react';
+import MUIAccordion from '@material-ui/core/Accordion';
+import MUIAccordionSummary from '@material-ui/core/AccordionSummary';
+import MUIAccordionDetails from '@material-ui/core/AccordionDetails';
+import MUIAccordionActions from '@material-ui/core/AccordionActions';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { IconButton } from '@material-ui/core';
+
+type AccordionHeadProps = {|
+  children: React.Node,
+  actions?: Array<React.Node>,
+  expandIcon?: React.Node,
+|};
+
+/**
+ * The header of an accordion section.
+ * Based on Material-UI AccordionSummary.
+ */
+export const AccordionHeader = (props: AccordionHeadProps) => {
+  return (
+    <MUIAccordionSummary
+      expandIcon={
+        props.expandIcon || (
+          <IconButton size="small">
+            <ExpandMoreIcon />
+          </IconButton>
+        )
+      }
+    >
+      <div style={{ flexGrow: 1 }}>{props.children}</div>
+      {props.actions && (
+        <div style={{ flexGrow: 0, alignSelf: 'center' }}>{props.actions}</div>
+      )}
+    </MUIAccordionSummary>
+  );
+};
+
+type AccordionBodyProps = {|
+  children: React.Node,
+|};
+
+/**
+ * The collapsible contents of an accordion section.
+ * Based on Material-UI AccordionDetails.
+ */
+export const AccordionBody = (props: AccordionBodyProps) => {
+  return <MUIAccordionDetails>{props.children}</MUIAccordionDetails>;
+};
+
+type AccordionActionsProps = {|
+  actions: Array<React.Node>,
+  secondaryActions?: Array<React.Node>,
+|};
+
+/**
+ * The footer of an accordion section, used to provide
+ * actions specific to the accordion contents.
+ * Based on Material-UI AccordionActions.
+ */
+export const AccordionActions = (props: AccordionActionsProps) => {
+  const accordionActions = props.secondaryActions ? (
+    <React.Fragment>
+      <div key="secondary-actions">{props.secondaryActions}</div>
+      <div key="actions">{props.actions}</div>
+    </React.Fragment>
+  ) : (
+    props.actions
+  );
+
+  return <MUIAccordionActions>{accordionActions}</MUIAccordionActions>;
+};
+
+type AccordionProps = {|
+  // AccordionSummary, AccordionBody and/or AccordionDetails
+  children: React.Node,
+
+  defaultExpanded?: boolean,
+  disabled?: boolean,
+
+  // If `true`, renders body only if accordion is open
+  costlyBody?: boolean,
```

Comment:
Ah that's great! I was thinking of this kind of property. For the sprite editor or maybe actually most editors, it might be worth not rendering the body of a section if it's collapsed.
It's harder to animate, but still better if it's bringing better performances :)